typed-racket-lib/typed-racket/optimizer/float-complex.rkt

```diff
@@ -82,18 +92,16 @@
     (define c1 (stx-car cs))
     (define o-nf (as-non-float o))
     (define c1-nf (as-non-float c1))
-    (if (or o-nf c1-nf)
-        ;; can't convert those to floats just yet, or may change
-        ;; the result
-        (let ([new-o (mark-as-non-float
-                      (quasisyntax/loc this-syntax
-                        (#,op #,(or o-nf o) #,(or c1-nf c1))))])
-          (if (stx-null? (stx-cdr cs))
-              new-o
-              (loop new-o
-                    (stx-cdr cs))))
-        ;; we've hit floats, can start coercing
-        (n-ary->binary this-syntax unsafe (cons #`(real->double-flonum #,(or o-nf o)) cs)))))
+    (cond
+      [(or o-nf c1-nf)
+       ;; can't convert those to floats just yet, or may change
+       ;; the result
+       (define new-o
+         (mark-as-non-float (quasisyntax/loc this-syntax
+                              (#,op #,(or o-nf o) #,(or c1-nf c1)))))
+       (if (stx-null? (stx-cdr cs)) new-o (loop new-o (stx-cdr cs)))]
```

Comment:
This should not be collapsed to one line with the `fmt` PR that @ 9999years is making.